pkg/reconciler/pipeline/dag/dag_test.go

```diff
@@ -219,22 +235,43 @@ func TestBuild_JoinMultipleRoots(t *testing.T) {
 }
 
 func TestBuild_FanInFanOut(t *testing.T) {
-	a := v1beta1.PipelineTask{Name: "a"}
+	a := v1beta1.PipelineTask{
+		Name: "a",
+		TaskSpec: &v1beta1.EmbeddedTask{TaskSpec: v1beta1.TaskSpec{
+			Results: []v1beta1.TaskResult{{
+				Name: "result",
+			}},
+		}},
+	}
 	dDependsOnA := v1beta1.PipelineTask{
 		Name: "d",
-		Resources: &v1beta1.PipelineTaskResources{
-			Inputs: []v1beta1.PipelineTaskInputResource{{From: []string{"a"}}},
-		},
+		TaskSpec: &v1beta1.EmbeddedTask{TaskSpec: v1beta1.TaskSpec{
+			Results: []v1beta1.TaskResult{{
+				Name: "resultFromD",
+			}},
+		}},
+		Params: []v1beta1.Param{{
+			Value: *v1beta1.NewStructuredValues("$(tasks.a.results.result)"),
+		}},
 	}
 	eRunsAfterA := v1beta1.PipelineTask{
```

Comment:
nit: the var name suggests eRunsAfterA but the struct suggest it depends on the result of A